src/test/scala/org/constellation/domain/redownload/DownloadServiceTest.scala

```diff
@@ -0,0 +1,81 @@
+package org.constellation.domain.redownload
+
+import cats.effect.{ContextShift, IO}
+import org.constellation.{ConstellationExecutionContext, DAO, TestHelpers}
+import org.constellation.p2p.Cluster
+import org.mockito.{ArgumentMatchersSugar, IdiomaticMockito}
+import org.mockito.cats.IdiomaticMockitoCats
+import org.scalatest.{BeforeAndAfterEach, FreeSpec, Matchers}
+
+class DownloadServiceTest
+    extends FreeSpec
+    with Matchers
+    with IdiomaticMockito
+    with IdiomaticMockitoCats
+    with BeforeAndAfterEach
+    with ArgumentMatchersSugar {
+
+  implicit val cs: ContextShift[IO] = IO.contextShift(ConstellationExecutionContext.unbounded)
+
+  var cluster: Cluster[IO] = _
+  var downloadService: DownloadService[IO] = _
+  var redownloadService: RedownloadService[IO] = _
+  implicit var dao: DAO = _
+
+  override def beforeEach() = {
+    dao = TestHelpers.prepareMockedDAO()
+    cluster = dao.cluster
+    redownloadService = dao.redownloadService
+    downloadService = new DownloadService(redownloadService, cluster)
+
+    dao.blacklistedAddresses.clear shouldReturnF Unit
+    dao.transactionChainService.clear shouldReturnF Unit
+    dao.addressService.clear shouldReturnF Unit
+    dao.soeService.clear shouldReturnF Unit
+  }
+
+  "requestForFaucet" - {
+    "to be defined" ignore {}
```

Comment:
Should this be within the scope of the DownloadService?

Reply:
Yes, and we still missing the answer how the faucet should work, because the previous implementation has been draining the tokens from the node

Reply:
Well short answer is that we don't need this for Main Net. But draining from the node's balance is fine for the faucet